source/_integrations/spotify.markdown

```diff
@@ -113,6 +115,32 @@ script:
 
 The `media_content_id` value can be obtained from the Spotify desktop app by clicking on the more options ("...") next to the album art picture, selecting "Share" and then "Copy Spotify URI" or "Copy Playlist Link" (also available in the Spotify phone and web app).
 
+The `media_content_id` can be a URL (e.g. `https://open.spotify.com/playlist/37i9dQZF1DWVFeEut75IAL`) or the Spotify URI string (e.g. `spotify:playlist:37i9dQZF1DWVFeEut75IAL`). (play_media)[https://www.home-assistant.io/integrations/media_player/#service-media_playerplay_media] requires the correct corresponding `media_content_type`.
+
+## Selecting output source
+
+To play media Spotify needs a device selected for audio output known as the `source`. Currently, this integration cannot initiate playback to a device not already available from the Spotify API. You need to first use Spotify on your smart speaker or phone to populate the source list.
```

Comment:
This makes it sound like this is an integration issue, which it is not. It is a limitation of the Spotify API.

> You need to first use Spotify on your smart speaker or phone to populate the source list.

That won't work as intuitively as described. It will take quite some time before that becomes visible. Instead, one should better be using the capabilities of the device and its specific integrations provided to use Spotify.

Reply:
Your comment here makes me think I didn't write this clearly so yes I will amend it.

> That won't work as intuitively as described...

I open the Spotify app and select a device from there. Alternatively I tell Google Home to cast Spotify. Both ways make the device available intermediately. Are you suggesting some other way like HACS Spotcast or was my description not clear?


Good point about where the limitation comes from. Perhaps it should say "Currently, the Spotify API cannot initiate playback to a device not already known to the Spotify and in the source list"